### PR TITLE
chore: Porting over tests from `syntax-transformer` and disabling those that don't current work.

### PR DESCRIPTION
**Pull request by d-jeffery**
## Description

* Moved over test suite from `syntax-transformer`, and disabled non-syntax related test cases.
* Modified test suite to expect throws from cases with 1 or more errors
* Minor modifications to grammar for syntax parsing of valid model
* Small modification to `OpenFgaDslSyntaxMultipleError` to preserve original error metadata

## References

- Closes https://github.com/openfga/language/issues/18

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: pkg/js/transformer/_testcases.ts [hunk 1]
```diff
@@ -56,3 +57,17 @@ export function loadInvalidDslSyntaxTestCases(): InvalidDslSyntaxTestCase[] {
 
   return JSON.parse(jsonData.toString("utf8")) as InvalidDslSyntaxTestCase[];
 }
+
+
+interface DslSyntaxErrorTestCase {
```
Comment (rhamzeh) [inline pkg/js/transformer/_testcases.ts:62]: Use the `InvalidDslSyntaxTestCase` interface. You don't need this

File: pkg/js/package.json [hunk 2]
```diff
@@ -34,13 +34,17 @@
...
   "devDependencies": {
     "@types/jest": "^29.5.1",
+    "@types/js-yaml": "^4.0.5",
     "@types/node": "^20.4.5",
     "@typescript-eslint/eslint-plugin": "^6.2.0",
     "@typescript-eslint/parser": "^6.2.0",
     "eslint": "^8.46.0",
     "eslint-config-prettier": "^8.9.0",
     "eslint-plugin-import": "^2.28.0",
+    "flatted": "^3.2.7",
```
Comment (rhamzeh) [inline pkg/js/package.json:44]: Where is this used?

Comment (d-jeffery) [reply pkg/js/package.json:44]: Good catch. Not used anymore.

File: pkg/js/transformer/_testcases.ts (hunk 1 above)
Comment (d-jeffery) [reply pkg/js/transformer/_testcases.ts:62]: Gotcha.

Comment (d-jeffery) [reply pkg/js/transformer/_testcases.ts]: Actually, based on the current structure of the test data, it was being used to deal with the current yaml structure, that is undecided on. Primarily I was using this in the tests to count how many error cases were expected.

https://github.com/openfga/language/blob/cc5a21078e403cf44614079078eacda5da27eed4/tests/data/dsl-syntax-validation-cases.yaml#L263-L281

Comment (rhamzeh) [reply pkg/js/transformer/_testcases.ts]: Extend the previous interface. In the end all invalid tests need to have these.

**Review by rhamzeh — APPROVED** [review]
@d-jeffery this PR closes part of the related issue. If you want to close it, please create a new one that is about the actual validation, as that part is still to be done.

**Pull request merged**